src/Objects/_PVector.js

```diff
@@ -0,0 +1,283 @@
+             len > 1? this.normalize(target).mult(length) :
+             argsErr('setMag', len, 1);
+    },
+    rotate: function(angle) {
+      const prev_x = this.x,
+            c = Math.cos(angle),
+            s = Math.sin(angle);
+      this.x = c*this.x - s*this.y;
+      this.y = s*prev_x + c*this.y;
+      return this;
+    },
+    dist: function(v1, v2) {
+      return v2? _PVector.dist(v1, v2) : _PVector.dist(this, v1);
+    },
+    distSq: function(v1, v2) {
+      return v2? _PVector.distSq(v1, v2) : _PVector.distSq(this, v1);
+    },
+    dot: function(v, y, z) {
+      const len = arguments.length;
+      return len === 1? _PVector.dot(this, v) :
+             len === 2? _PVector.dot(v, y) :
+             len > 2? this.x*v + this.y*y + this.z*z :
+             argsErr('dot', len, 1);
+    },
+    cross: function(v1, v2, target) {
+      return target? _PVector.cross(v1, v2, target)
+                   : _PVector.cross(this, v1, v2);
+    },
+    lerp: function() {
+      var x, y, z, amt;
+      const len = arguments.length;
+      if ((len | 1) === 1) { argsErr('lerp', len, 2); }
+      if (len === 2) { // given vector and amt
+        const v = arguments[0];
+        x = v.x;
+        y = v.y;
+        z = v.z;
+        amt = arguments[1];
+      } else if (len === 3) { // given vector 1, vector 2 and amt
+        return _PVector.lerp(arguments[0], arguments[1], arguments[2]);
+      } else { // given x, y, z and amt
+        x = arguments[0];
+        y = arguments[1];
+        z = arguments[2];
+        amt = arguments[3];
+      }
+      this.x = lerp(this.x, x, amt);
+      this.y = lerp(this.y, y, amt);
+      this.z = lerp(this.z, z, amt);
+      return this;
+    },
+    add: function(v, y, z) {
+      if (y instanceof _PVector) { return _PVector.add(v, y, z); }
+      const len = arguments.length;
+      if (len === 1) {
+        //_PVector.add(this, v, this);
+        this.x += v.x;
+        this.y += v.y;
+        this.z += v.z;
+      } else if (len > 1) {
+        this.x += v;
+        this.y += y;
+        if (len > 2) { this.z += z };
+      } else { argsErr('add', len, 1); }
+      return this;
+    },
+    sub: function(v, y, z) {
+      if (y instanceof _PVector) { return _PVector.sub(v, y, z); }
+      const len = arguments.length;
+      if (len === 1) {
+        this.x -= v.x;
+        this.y -= v.y;
+        this.z -= v.z;
+      } else if (len > 1) {
+        this.x -= v;
+        this.y -= y;
+        if (len > 2) { this.z -= z; }
+      } else { argsErr('sub', len, 1); }
+      return this;
+    },
+    mult: function(v, n, target) {
+      const len = arguments.length;
+      if (len === 1) {
+        if (typeof v === 'number') {
+          this.x *= v;
+          this.y *= v;
+          this.z *= v;
+        } else {
+          this.x *= v.x;
+          this.y *= v.y;
+          this.z *= v.z;
+        }
+        return this;
+      } else if (len > 1) { return _PVector.mult(v, n, target); }
```

Comment:
Yes, it still does that? I'm only suggesting rewriting it such that we first error, if we need to, then we resolve the len===1 exactly at it happens now, ending in `return this`, but because of that return, we don't need the `else if` syntax for the `len>1` case.